### PR TITLE
Inspector: Fix texture preview in WebGPU

### PR DESCRIPTION
**Pull request by Popov72**
See https://forum.babylonjs.com/t/texture-preview-does-not-work-in-the-latest-webgpu-playground/53729

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/ShadersWGSL/lod.fragment.fx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+var textureSamplerSampler: sampler;
+var textureSampler: texture_2d<f32>;
+
+uniform lod: f32;
+uniform gamma: bool;
+
+@fragment
+fn main(input: FragmentInputs) -> FragmentOutputs {
+	fragmentOutputs.color = textureSampleLevel(textureSampler, textureSamplerSampler, vUV, uniforms.lod);
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/lod.fragment.fx:13]:
```suggestion
	fragmentOutputs.color = textureSampleLevel(textureSampler, textureSamplerSampler, input.vUV, uniforms.lod);

```

File: packages/dev/core/src/ShadersWGSL/lodCube.fragment.fx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+var textureSamplerSampler: sampler;
+var textureSampler: texture_3d<f32>;
+
+uniform lod: f32;
+uniform gamma: bool;
+
+@fragment
+fn main(input: FragmentInputs) -> FragmentOutputs {
+    let uv = vUV * 2.0 - 1.0;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/lodCube.fragment.fx:13]:
```suggestion
    let uv = input.vUV * 2.0 - 1.0;

```

File: packages/dev/core/src/ShadersWGSL/lodCube.fragment.fx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    #ifdef NEGATIVEY
+        fragmentOutputs.color = textureSampleLevel(textureSampler, textureSamplerSampler, vec3f(uv.y,-1.001,uv.x), uniforms.lod);
+    #endif
+    #ifdef POSITIVEZ
+        fragmentOutputs.color = textureSampleLevel(textureSampler, textureSamplerSampler, vec3f(uv,1.001), uniforms.lod);
+    #endif
+    #ifdef NEGATIVEZ
+        fragmentOutputs.color = textureSampleLevel(textureSampler, textureSamplerSampler, vec3f(uv,-1.001), uniforms.lod);
+    #endif
+    if (!gamma) {
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/lodCube.fragment.fx:32]:
```suggestion
    if (!uniforms.gamma) {
```

File: packages/dev/core/src/ShadersWGSL/lod.fragment.fx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    if (!gamma) {
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/lod.fragment.fx:14]:
```suggestion
    if (!uniforms.gamma) {
```

**Review by noname0310 — COMMENTED** [review]
I just casually looked at it and noticed these mistakes

**Comment by Popov72** [comment]
I agree with your changes, but it was working in my testing... Let me have a look!

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15626/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15626/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15626/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15626/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15626/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15626/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15626/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15626/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15626/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15626/merge/testResults/webgl2playwright/index.html

**Pull request merged**